pkg/apiserver/apiserver.go

```diff
@@ -250,7 +285,11 @@ func AddApisWebService(s runtime.NegotiatedSerializer, container *restful.Contai
 // Adds a service to return the supported versions, preferred version, and name
 // of a group. E.g., a such web service will be registered at /apis/extensions.
 func AddGroupWebService(s runtime.NegotiatedSerializer, container *restful.Container, path string, group unversioned.APIGroup) {
-	groupHandler := GroupHandler(s, group)
+	ss := s
+	if keepUnversioned(group.Name) {
```

Comment:
Comments with each of these and a test to verify no regression, approach looks fine